macros/misc/randomNamesPronouns.pl

```diff
@@ -0,0 +1,617 @@
+	Foster  Garcia    Gomez      Gonzales Gray    Green    Gutierrez Hall
+	Harris  Hernandez Hill       Howard   Hughes  Jackson  James     Jimenez
+	Johnson Jones     Kelly      Kim      King    Lee      Lewis     Long
+	Lopez   Martin    Martinez   Mendoza  Miller  Mitchell Moore     Morales
+	Morgan  Morris    Murphy     Myers    Nelson  Nguyen   Ortiz     Parker
+	Patel   Perez     Peterson   Phillips Pina    Price    Ramirez   Ramos
+	Reed    Reyes     Richardson Rivera   Roberts Robinson Rodriguez Rogers
+	Ross    Ruiz      Sanchez    Sanders  Scott   Smith    Stewart   Sullivan
+	Taylor  Thomas    Thompson   Torres   Turner  Walker   Ward      Watson
+	White   Williams  Wilson     Wood     Wright  Young
+);
+
+# need to have the keys sorted to have a consistent set of names for a given seed, so
+# we build an array of the firstnames sorted.
+my @firstnames;
+
+sub _randomNamesPronouns_init {
+	@firstnames = lex_sort(keys(%$namespronouns));
+}
+
+=head2 randomPerson
+
+Returns a person as a Person object from a list in the macro.
+
+=head3 Examples and options
+
+=over
+
+=item * No arguments returns a single random person
+
+Example
+
+  randomPerson()
+
+=item * C<S<< n => k >>> returns an array of k Person objects with unique names.
+
+Example
+
+  randomPerson(n=>5)
+
+returns an array of 5 Person objects with unique names.
+
+=item * C<S<< names => arrayref >>> returns either a single Person or array of Persons
+from the given names
+
+If the arrayref is in the form of C<[['name1','pronoun1'],['name2','pronoun2'],...]> then
+a person selected randomly from the given list is returned.
+
+Example:
+
+  $a = randomPerson(names => [['Bart','he'], ['Lisa','she'], ['Matty','they']]);
+
+Alternatively, each person name/pronoun can be set as an arrayref.  For example the above can be written:
+
+  $a = randomPerson(names => [
+    { name => 'Bart', pronoun => 'he' },
+    { name => 'Lisa', pronoun => 'she' },
+    { name => 'Matty', pronoun => 'they' } ]);
+
+If the pronoun is missing using either arrayrefs or hashrefs, then a pronoun is determined randomly.
+Each of the following are legal
+
+  $p1 = randomPerson( names => ['Larry', 'Moe', 'Curly']);
+  $p2 = randomPerson( names => ['Larry', ['Moe', 'he'], 'Curly']);
+  $p3 = randomPerson( names => [{ name => 'Larry'}, { name => 'Moe' }, { name => 'Curly' }]);
+  $p4 = randomPerson( names => [{ name => 'Larry'}, { name => 'Moe', pronoun => 'he' }, { name => 'Curly' }]);
+
+And the option C<n> can be used to return an array with that number of persons.
+
+  @p = randomPerson(n => 2, names => [['Bart','he'], ['Lisa','she'], ['Matty','they']]);
+
+=back
+
+=cut
+
+sub randomPerson {
+	my %options = (n => 1, @_);
+
+	# if the names are passed in.
+	if ($options{names}) {
+		my @persons;
+		for my $p (@{ $options{names} }) {
+			if (ref $p ne 'HASH') {
+				push(@persons,
+					Person->new(name => $p->[0] // $p, pronoun => $p->[1] // list_random('he', 'she', 'they')));
```

Comment:
`$p->[0] // $p` for the name is not good.  The only reason this works is because macros do not enforce `strict refs`.  If `use strict` were properly in effect this would be an error in the case that `$p` is not an array reference.  My suggested `randomPerson` method in the other comment does this correctly.